stan/math/prim/functor/operands_and_partials.hpp

```diff
@@ -45,40 +67,42 @@ class ops_partials_edge<ViewElt, Op, require_st_arithmetic<Op>> {
   partials_t partials_;
   empty_broadcast_array<partials_t, inner_op> partials_vec_;
   static constexpr double operands_{0};
-  ops_partials_edge() {}
-  template <typename T>
-  explicit ops_partials_edge(T&& /* op */) noexcept {}
+  ops_partials_edge() = default;
+
+  template <typename T,
+            require_not_same_t<
+                std::decay_t<T>,
+                std::decay_t<ops_partials_edge<
+                    ViewElt, Op, require_st_arithmetic<Op>>>>* = nullptr>
+  constexpr explicit ops_partials_edge(T&& /* op */) noexcept {}
 
   /**
    * Get the operand for the edge. For doubles this is a compile time
    * expression returning zero.
    */
-  static constexpr double operand() noexcept { return 0.0; }
+  static constexpr auto operand() noexcept { return static_cast<double>(0.0); }
```

Comment:
Is there a particular benefit for this change? Won't the type deduction be the same?

Reply:
No there is not, not sure why I changed this i'll put it back